mathesar/api/db/viewsets/columns.py

```diff
@@ -207,6 +208,7 @@ def dependents(self, request, pk=None, table_pk=None):
 
     def destroy(self, request, pk=None, table_pk=None):
         column_instance = self.get_object()
+        remove_preview_columnId_from_cache(columnId=column_instance.id)
```

Comment:
You don't want to do selective state updates. Only full resets via `reset_reflection`. Well, not completely full. `reset_reflection` takes a database as argument.